scipy/spatial/tests/test_spherical_voronoi.py

```diff
@@ -232,67 +230,13 @@ def test_single_hemisphere_handling(self):
-            spherical_voronoi.SphericalVoronoi(points)
-
-    @pytest.mark.parametrize("n", [8, 15, 21])
-    @pytest.mark.parametrize("radius", [0.5, 1, 2])
-    @pytest.mark.parametrize("fraction", [0, 0.6])
+    @pytest.mark.parametrize("n", [1, 2, 10])
     @pytest.mark.parametrize("center", [(0, 0, 0), (1, 2, 3)])
-    def test_circular_input(self, n, radius, fraction, center):
-        # generate a random circle
-        rng = np.random.RandomState(0)
-        U = Rotation.random(random_state=rng).as_matrix()
-        h = radius * fraction
-        circle_radius = np.sqrt(radius**2 - h**2)
+    def test_rank_deficient(self, n, center):
         thetas = np.linspace(0, 2 * np.pi, n, endpoint=False)
-        points = np.vstack([circle_radius * np.sin(thetas),
-                            circle_radius * np.cos(thetas),
-                            h * np.ones(n)]).T
-        points = points @ U
-
-        # calculate Spherical Voronoi diagram
-        sv = SphericalVoronoi(points + center, radius=radius, center=center)
-
-        # each region must have 4 vertices
-        region_sizes = np.array([len(region) for region in sv.regions])
-        assert (region_sizes == 4).all()
-        regions = np.array(sv.regions)
-
-        # vertices are those between each pair of input points + north and
-        # south poles
-        vertices = sv.vertices - center
-        assert len(vertices) == n + 2
-
-        # verify that north and south poles are orthogonal to circle on which
-        # input points lie
-        poles = vertices[n:]
-        midpoint = np.array([0, 0, h]) @ U
-        assert np.abs(np.dot(points - midpoint, poles.T)).max() < 1E-10
-
-        # test arc lengths to forward and backward neighbors
-        for point, region in zip(points, sv.regions):
-            cosine = np.dot(vertices[region] - midpoint, point - midpoint)
-            sine = np.linalg.norm(np.cross(vertices[region] - midpoint,
-                                           point - midpoint), axis=1)
-            arclengths = circle_radius * np.arctan2(sine, cosine)
-            assert_almost_equal(arclengths[[0, 2]], circle_radius * np.pi / n)
-
-        # test arc lengths to poles
-        for point, region in zip(points, sv.regions):
-            cosine = np.dot(vertices[region], point)
-            sine = np.linalg.norm(np.cross(vertices[region], point), axis=1)
-            arclengths = np.arctan2(sine, cosine)
-            angle = np.arcsin(h / radius)
-            assert_almost_equal(arclengths[1], np.pi / 2 - angle)
-            assert_almost_equal(arclengths[3], np.pi / 2 + angle)
-
-        regions = sv.regions.copy()
-        sv.sort_vertices_of_regions()
-        assert regions == sv.regions
+        points = np.vstack([np.sin(thetas), np.cos(thetas), np.zeros(n)]).T
+        with pytest.raises(ValueError, match="Rank of input points"):
+            spherical_voronoi.SphericalVoronoi(points)
```

Comment:
`center` never gets used from the parameters above now, right?

These are all equatorial cases centered at the origin now?

Reply:
Good catch!

That was unintentional. I merged the `test_rank_deficient` and `test_circular_input` methods, and kept the wrong instantiation of `SphericalVoronoi`.